### PR TITLE
Add a "safe" constructor which isn't const constructable

### PR DESCRIPTION
**Pull request by hugovdm**
WIP (do not submit!)

This new constructor does some canonicalization of the input fields at construction time.

- We could also support all of this change with just the unnamed constructor, at the cost of making the getters a bit slower (by doing this processing at get-time. I think lowercase&uppercase are the most expensive new operations).
- Reporting errors might be nice. Optionally throw an exception if garbage is given? Selected by boolean named parameter?
- Documentation still needs improvement, leaving that until the code is considered good.

### CODE REVIEW

File: lib/ui/window.dart [hunk 1]
```diff
@@ -150,8 +156,30 @@ class Locale {
...
+  /// even if the input is garbage.)
+  ///
+  /// In its current state, this code throws away bad tags. Bad language tags
+  /// become "und" (undefined). Bad region tags are discarded giving a
+  /// regionless Locale.
+  ///
+  /// Other options we could consider is throwing exceptions: it would be nice
+  /// to provide feedback to calling code shoud it want feedback, so that it
+  /// could make its own decision on how to deal with garbage.
+  Locale.safe(String language, { String region }) :
```
Comment (Hixie) [inline lib/ui/window.dart]: what is the use case for this?

Comment (hugovdm) [reply lib/ui/window.dart]: For me, my primary purpose of this pull request is to decide whether we'd like the getters to call toLowerCase() and toUpperCase() - in which case the unnamed constructor becomes case insensitive.

I'll come back to this suggestion/request on Monday, adjusting it in one of two directions:
(a) proposing a change to make the unnamed constructor take care of all of this,
(b) making a Locale.parse proposal to at least get this functionality submitted, following which we could still revisit the question "should the unnamed constructor perhaps be case insensitive?"

**Comment by hugovdm** [comment]
Will make an entirely fresh pull request. I'm rebasing anyway, and might as well rename the branch.

File: lib/ui/window.dart (hunk 1 above)
Comment (hugovdm) [reply lib/ui/window.dart]: The purpose of this constructor was to have a constructor that won't ever fail, while producing a "syntactically valid" (as per Unicode LDML specification) Locale instance - i.e. a workaround for the question of "what counts as misuse" and my desire to add some syntax-checking asserts to the existing const nameless constructor.

**Pull request closed without merging**